lib/Target/X86/X86FrameLowering.cpp

```diff
@@ -549,54 +548,69 @@ MachineInstr *X86FrameLowering::emitStackProbeInline(
       JoinReg = InProlog ? X86::RCX : MRI.createVirtualRegister(RegClass),
       ProbeReg = InProlog ? X86::RCX : MRI.createVirtualRegister(RegClass);
 
-  // If in the prolog, push RCX and RDX. If not, copy RAX to a virtual reg.
+  // SP-relative offsets where we can save RCX and RDX.
+  int64_t RCXShadowSlot = 0;
+  int64_t RDXShadowSlot = 0;
+
+  // If inlining in the prolog, save RCX and RDX.     
+  // Future optimization: don't save or restore if not live in.
   if (InProlog) {
-    BuildMI(&MBB, DL, TII.get(X86::PUSH64r)).addReg(X86::RCX);
-    BuildMI(&MBB, DL, TII.get(X86::PUSH64r)).addReg(X86::RDX);
+    // Compute the offsets.
+    X86MachineFunctionInfo *X86FI = MF.getInfo<X86MachineFunctionInfo>();
+    const int64_t CalleeSaveSize = X86FI->getCalleeSavedFrameSize();
+    const bool HasFP = hasFP(MF);
+    RCXShadowSlot = 8 + CalleeSaveSize + (HasFP ? 8 : 0);
+    RDXShadowSlot = 16 + CalleeSaveSize + (HasFP ? 8 : 0);
```

Comment:
Might be worth a comment for the magic 8/16 at the start here.  I presume it's the return address in the first case and additionally the rcx slot in the second case?  Might want to just define RDXShadowSlot as RCXShadowSlot + 8.